src/sdam/monitor.rs

```diff
@@ -55,13 +52,13 @@ impl Monitor {
             .heartbeat_freq
             .unwrap_or(DEFAULT_HEARTBEAT_FREQUENCY);
 
-        while let Some(server) = self.server.upgrade() {
+        while let Some(_server) = self.server.upgrade() {
```

Comment:
We want to make sure we can upgrade the weak reference to the server to a strong reference, since being unable to indicates that the server was dropped from the cluster, but we don't actually need the server itself for the body of the loop.